### ISSUE TITLE
Graph API inserts not visible within transaction

### ISSUE THREAD

**Issue by vertex-github** [orientechnologies/orientdb#3599]
OrientDB v2.02

We expected updates within the transaction to be visible to the current thread:

````java
private void testDeleteGraphAPI( )
   {
      User user = new User( "test@test.com" );
      OrientGraph graph = orientGraphFactory.getTx( );

      OrientVertex v;
      try
      {
         graph.begin();

         v = graph.addVertex( "class:User" );
         v.setProperty( "emailAddress", user.getEmailAddress( ) );

         user.setIdentity( v.getIdentity( ) );

         assertEquals( 1, graph.countVertices( ) ); // <--- this actually returns 0

         graph.removeVertex( v );

         assertEquals( 0, graph.countVertices( ) );
      }
      finally
      {
         graph.shutdown();
      }
   }
````

**Comment by pjweisberg** [comment]
I've stumbled on the same issue, or at least a related one.  One of the things I'm trying to do is find a vertex that may have just been created.  It works as expected in tests that use an in-memory database, but fails when using a remote database.

    @Test
    public void testFindCreated() throws Exception {

        //Fails only when using a remote address.  Works when using "memory:df".
        OrientGraphFactory orientGraphFactory = new OrientGraphFactory("remote:endevdf04/df");
        OrientGraph graph = orientGraphFactory.getTx( );
        
        OrientVertex vertex = graph.addVertex("class:DFProvider");
        vertex.setProperty("npi", "1234");
        
        Vertex retrievedVertex = graph.getVertexByKey("DFProvider.npi", "1234");
        assertNotNull(retrievedVertex);
    }

**Comment by lvca** [comment]
This is documented and how remote works. It will be addressed with 3.0.
